community/modules/compute/gke-job-template/variables.tf

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+variable "name" {
+  description = "The name of the job."
+  type        = string
+  default     = "my-job"
+}
+
+variable "node_count" {
+  description = "How many nodes the job should run in parallel."
+  type        = number
+  default     = 1
+}
+
+variable "command" {
+  description = "A list of strings that will be joined to create the job command."
+  type        = list(string)
+  default     = ["hostname"]
+}
+
+variable "image" {
+  description = "The container image the job should use."
+  type        = string
+  default     = "debian"
+}
+
+variable "node_pool_name" {
+  description = "The name of the node pool on which to run the job. Can be populated via `use` feild."
+  type        = string
+  default     = null
+}
+
+variable "cpu_per_node" {
+  description = "The number of CPUs per node. Used to claim whole nodes. Generally populated from gke-node-pool via `use` field."
+  type        = number
+  default     = null
+}
+
+variable "tolerations" {
+  description = "value"
+  type = list(object({
+    key      = string
+    operator = string
+    value    = string
+    effect   = string
+  }))
+  default = [
+    {
+      key      = "user-workload"
+      operator = "Equal"
+      value    = "true"
+      effect   = "NoSchedule"
+    }
+  ]
+}
+
+variable "machine_family" {
+  description = "The machine family to use in the node selector (example: `n2`). If null then machine family will not be used as selector criteria."
```

Comment:
Is that not part of the node pool?

Reply:
As description explains, this is used for the node selector. 

Say I have 3 node pools that all use different sizes of C2 machines. I could use this variable to include all of them. This is meant to be a user facing variable, where as the node pool names references a specific node pool. For example you could have:

```
  - id: job-template
    source: ./community/modules/compute/gke-job-template
    settings:
      requested_cpu_per_pod: 4
      machine_family: c2
    outputs: [instructions]
```